CONTRIBUTING.md

```diff
@@ -43,6 +43,19 @@ querying network status and applying network states.
 
 - `./tests/` - Contains tests for unit and integration tests.
 
+## Installation and Compilation:
+
+- [Fork](https://github.com/nmstate/nmstate.git) and git clone to local
+-[Install rust and Cargo for the development of nmstate](https://doc.rust-lang.org/cargo/getting-started/installation.html) or `apt-get install cargo`
+- Change your current directory to the nmstate directory `cd nmstate`
+- Install nmstate `cargo install nmstatectl`
+This will download the necessary dependencies and build the nmstate library and command-line tool
+- Build and compile the packages using `cargo build`
+This will compile the source code and generate the executable files
+- Run unit test to ensure functionality of the  nmstate library and tools `cargo test`
+- View networking state in yaml format `nmstatectl show`
+- Check version `nmstatectl --version`
```

Comment:
To do this, you need to install it first. You have not installed anything from the repository in any step.